event/datadeal/data_verification_event.go

```diff
@@ -1 +1,216 @@
 package datadeal
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"strconv"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/medibloc/panacea-core/v2/x/datadeal/types"
+	oracletypes "github.com/medibloc/panacea-core/v2/x/oracle/types"
+	"github.com/medibloc/panacea-doracle/config"
+	"github.com/medibloc/panacea-doracle/crypto"
+	"github.com/medibloc/panacea-doracle/event"
+	"github.com/medibloc/panacea-doracle/ipfs"
+	"github.com/medibloc/panacea-doracle/panacea"
+	"github.com/medibloc/panacea-doracle/sgx"
+	"github.com/medibloc/panacea-doracle/validation"
+	log "github.com/sirupsen/logrus"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
+var _ event.Event = (*DataVerificationEvent)(nil)
+
+type DataVerificationEvent struct {
+	reactor reactor
+}
+
+type reactor interface {
+	GRPCClient() *panacea.GrpcClient
+	EnclaveInfo() *sgx.EnclaveInfo
+	OracleAcc() *panacea.OracleAccount
+	OraclePrivKey() *btcec.PrivateKey
+	Config() *config.Config
+	QueryClient() *panacea.QueryClient
+	Ipfs() *ipfs.Ipfs
+}
+
+func NewDataVerificationEvent(r reactor) DataVerificationEvent {
+	return DataVerificationEvent{r}
+}
+
+func (d DataVerificationEvent) GetEventType() string {
+	return "message"
+}
+
+func (d DataVerificationEvent) GetEventAttributeKey() string {
+	return "action"
+}
+
+func (d DataVerificationEvent) GetEventAttributeValue() string {
+	return "'SellData'"
+}
+
+func (d DataVerificationEvent) EventHandler(event ctypes.ResultEvent) error {
```

Comment:
I think oracle should be vote if there are no special problem.
For example, if oracle want to get deal, but deal is not exist, then oracle shoud be voted `no`.
But if oracle want to get deal, but received another error(exclude the not exist deal error), oracle should not vote.

Reply:
It makes sense. 
We have to make `EventHandler` to vote by judging whether the data is correct or not.

> For example, if oracle want to get deal, but deal is not exist, then oracle shoud be voted no.

The former case gyuguen mentioned is definitely invalid data sale. No one can sell their data to non-existing deal.

Reply:
Right. 
I think oracle cannot verify data if not eixst deal. 
So oracle votes no.

Does @H4NLee think when this case oracle should not voted?

Reply:
I meant that if there is no deal (when getting deal, got `NotExist` err), oracle should vote `NO`. 
The data sale for selling data on not existing deal should also be considered invalid I think.

Reply:
Aha, I understood your answer.
Thank you

Reply:
In addition to the case gyuguen said above, let me take a look at other cases.

Reply:
So the if any errors emerged, the oracle votes `NO`. Am I right?

Reply:
Maybe not. 

> For example, if oracle want to get deal, but deal is not exist, then oracle shoud be voted no.
But if oracle want to get deal, but received another error(exclude the not exist deal error), oracle should not vote.

As gyuguen said, we need to decide to vote or not depending on the errors. That's why I'm taking look at the cases.